scenario-compiler/src/main/scala/pl/touk/nussknacker/engine/spel/Typer.scala

```diff
@@ -484,16 +491,22 @@ private[spel] class Typer(
   private def operationOnTypesValue[A, B, R](left: TypingResult, right: TypingResult, fallbackType: TypingResult)(
       op: (A, B) => Validated[ExpressionParseError, R]
   ): TypingR[TypingResult] =
-    (for {
-      leftValue  <- left.valueOpt
-      rightValue <- right.valueOpt
-      res = op(leftValue.asInstanceOf[A], rightValue.asInstanceOf[B])
-    } yield {
-      res
-        .map(Typed.fromInstance)
-        .map(valid)
-        .valueOr(invalid(_, fallbackType))
-    }).getOrElse(valid(fallbackType))
+    (left.valueOpt, right.valueOpt) match {
+      case (Some(leftValue), Some(rightValue)) =>
+        Try(op(leftValue.asInstanceOf[A], rightValue.asInstanceOf[B])) match {
+          case Failure(a: ArithmeticException) =>
+            logger.debug(s"Arithmetic exception during op, using fallback. $a")
+            valid(fallbackType)
```

Comment:
Why this is valid?

Reply:
BigDecimal division frequently throws "Non-terminating decimal expansion", I think using the fallback (typedclass BigDecimal without value) is ok, but perhaps there's a way to round the result or something, i'll check

Reply:
dividing with RoundingMode.HALF_EVEN (consistent with SpEL evaluation) avoids this exception